microsoft/react-native-test-app
# fix(android): fix switching between dev server and embedded JS bundle

### Description

Fix switching between dev server and embedded JS bundle.

Resolves #2119.

### Platforms affected


### Test plan

1. Build the Android bundle and app:
    ```
    cd example
    yarn android
    ```
2. In a separate terminal, start the dev server: `yarn start`
3. Tap on "App"
4. Open the dev menu (Ctrl/⌥+m) and tap on "Load embedded JS bundle"
5. Verify that this navigates back to the Home activity
6. Tap on "App" again
7. You should get a dialog about missing JS bundle
8. From the … menu in the top right, select "Load from dev server"
9. Tap on "App"
10. Verify that it opens
11. Quit the app
12. Bundle the JS: `yarn build:android`
13. Rebuild the app (see step 1)
14. Tap on "App"
15. Open the dev menu (Ctrl/⌥+m) and tap on "Load embedded JS bundle"
16. Verify that this navigates back to the Home activity
17. Tap on "App" again
18. This will load the embedded bundle (indicated by the lack of dev menu)
19. Navigate back
20. From the … menu in the top right, select "Load embedded JS bundle" again
21. This should not crash
22. From the … menu in the top right, select "Load from dev server"
23. Tap on "App"
24. Verify that the dev menu can be opened (Ctrl/⌘+m)
